src/features/wallet/send/SendView.tsx

```diff
@@ -217,64 +204,55 @@ const SendView = ({ scanResult, sendType, hotspot, isSeller }: Props) => {
         )
       }
     } else {
-      const totalTxnAmount = balanceAmount.plus(fee)
-      // TODO balance compare/greater than/less than
-      const hasBalance =
-        totalTxnAmount.integerBalance <= (account?.balance?.integerBalance || 0)
+      let hasBalance = true
+      let isValidAddress = true
+      let isValidBalanceAmount = true
+      sendTransfers.forEach(({ address, balanceAmount, fee }) => {
```

Comment:
This is validating the amount for each payment in the array, but it should be adding up all the amounts and validating that the user has enough balance to pay the total

Reply:
Great catch, fixed here https://github.com/helium/hotspot-app/pull/479/commits/c37e5c463b597e613e27eaaa88ec71dacdd197e6

Reply:
This is still a little incorrect, on line 230 you have `totalSendAmount = totalSendAmount.plus(balanceAmount.plus(fee))` this should just be `totalSendAmount = totalSendAmount.plus(balanceAmount)` and then at the end outside the loop you just add the fee once, `totalSendAmount = totalSendAmount.plus(fee)`.

The fee is the transaction fee on the blockchain and its only one fee per transaction, which we are now using just one paymentV2 transcation with multiple payees and amounts.

Reply:
So in the end you add up, all the payment amount plus the one transaction fee

Reply:
Ohh very interesting, I did not realize the fee was calculated on a per-transaction basis (that could include 1 to many payments). Does the fee calculation itself have any dependence on the payment amounts? Wondering if the fee calculation should happen at the `SendForm` level instead of on each `SendDetailsForm` level.

For example, if you made 1 payment in a transaction for 5 HNT vs. 100 payments in a transaction for 50 HNT each, would both scenarios incur the same fee?

Reply:
Have made the immediate change [here](https://github.com/helium/hotspot-app/pull/479/commits/b2acc92aca47b91b5cfa353265f47570b08757ba), but lmk if it the above comment is still applicable (moving fee calculation up a level)

Reply:
Yeah thats a good point, the fee is based on the byte size of the transaction. So adding / removing any data from it will change it. Having more payee / payer pairs could definitely increase the fee so we would want to calculate it with that in mind.

Reply:
I think currently we show the fee under the amount, which isnt really correct when you have multiple payer / payee combos. We may need to move it. Im thinking possibly below the send button so that its always visible and you won't miss it due to having to scroll.

Reply:
Great suggestion - made the change to calculate 1 fee based on their entire transaction and show it under the submit button with https://github.com/helium/hotspot-app/pull/479/commits/d3d08a2aa540b13f46a46668514574dcd6e95846 and https://github.com/helium/hotspot-app/pull/479/commits/092235e5a3402c4c6ad94adf6c8b9c84b4a5d1bd

Reply:
Awesome this looks great